### DISCUSSION TITLE
显示原书用字

### DISCUSSION THREAD

**Discussion by ztl8702** [MindongLab/yngdieng#340] [产品功能 ideas]
## 背景

榕典为了减少初学者困惑，目前对数据来源作了用字修订。默认显示的不是原书的内容，而是经过编辑修订的“榕典编辑推荐用字”。

**Comment by ztl8702** [comment]
留意：
* 显示原书时， 冯书连读音隐藏

**Comment by ztl8702** [comment]
- [ ] TODO： 榕拼转冯书IPA的规则 @DDFShanyu  https://shimo.im/sheets/vXtJtdCtjHgDV83Y/MODOC

**Comment by ZySieng** [comment]
最近随着我们的来源增多（戚林、李书、dfd等），我对于显示原书这个功能有了更系统的设想，之前也没有完整地在github说过，就在这里说了。之前我们在KFC聊过一个显示原书的构想，那个我认为可以作废了。

### 总体设想：
**条目最上方的大字**，显示的是——推荐用字和榕拼
**每个来源的条目**，显示的是——原书用字和原书拼音
**每个来源的例句**，显示的是——推荐用字和榕拼
提供将“每个来源的例句”的用字从“推荐用字和榕拼”切换回“原书用字和原书拼音”的功能

### 图例：
![Screenshot_20201209_142023_com tencent mm](https://user-images.githubusercontent.com/32233404/101594975-96085580-3a2d-11eb-80cc-01037b24601f.jpg)


### 理据：
榕典总的来说是一部多来源的线上词典，统一用字和拼音也是为了方便搜索和查看，不是说就这么把原书内容给改掉了。
随着来源的增多，原来修改的方式已经不能够持续——原来的方法是我用{,}去改feng、li里面的数据。但是有了戚林、dfd、甚至是各地地方志之后，这种方法就没有用了。比如说houng1这个音，戚林写的是“烟”字，榕典推荐用字是“熏”，难不成还得把戚林八音改了？
加之，目前这样标记了出处，但部分内容又与原书不同，也不好。
因此，不能再这么改下去了，要作出改变，但是为了方便查询检索，统一用字和榕拼也不能舍弃。折衷的方法是，以推荐用字和榕拼为主，提供显示原书的功能。

用户搜出来的条目，**条目最上方的大字**，肯定是显示推荐用字和榕拼。
**每个来源的条目**，显示原书的用字和拼音——目前的榕典这里显示的依然是推荐用字，得改掉——因为像戚林八音这样的来源，我们不可能把他的用字给改掉。
为了方便用户查阅例句，我们将**每个来源里面的用字**默认显示为推荐用字，这里可以切换为原书用字，显示为条目旁白的一个**小按钮**。

### 方法：
技术不太了解，满讲的
以zingzeuID为纲，每个来源的条目都与ID作映射。
**条目最上方的大字**，直接显示zingzeuwords里面的用字和拼音
**每个来源的条目**，汉字显示的是feng、li里面{,}的逗号前的汉字，拼音则根据转化规则统一由榕拼转化而成
**每个来源的例句**，汉字显示的是feng、li里面{,}的逗号后的汉字，切换后，显示逗号前的

### 预先要解决的事情：
处理技术环节以外，有两件事情要紧
要将zingzeuID进行映射，譬如戚林，这个要早做
zingzeuwords内的用字和拼音，要和目前feng{,}修改后的保持一致。之前我们尝试过以zingzeuwords为纲来进行搜索，但是因为输入法词库有若干问题而废止了，现在要系统地检查一遍输入法的词库。我的看法是，先做冯版，看看输入法的用字、拼音能不能和修改后的冯版合得上，做一个系统性的筛查，不匹配的人工筛选出来。

**Reply by Guanchishan** [reply]
> zingzeuwords内的用字和拼音，要和目前feng{,}修改后的保持一致。之前我们尝试过以zingzeuwords为纲来进行搜索，但是因为输入法词库有若干问题而废止了，现在要系统地检查一遍输入法的词库。我的看法是，先做冯版，看看输入法的用字、拼音能不能和修改后的冯版合得上，做一个系统性的筛查，不匹配的人工筛选出来。

所以現在輸入法詞庫和馮版用字是脫鉤的嗎？亦即，在feng.txt改的字並沒有被完全的體現在zingzeuword，且目前feng與zingzeuword有潛在的用字不一樣的風險，需要進行大規模排查，是嗎？

**Reply by ztl8702** [reply]
> 在feng.txt改的字並沒有被完全的體現在zingzeuword，且目前feng與zingzeuword有潛在的用字不一樣的風險

正确。目前优先显示 feng 的“修改后”用字。

**Reply by Guanchishan** [reply]
> 每个来源的条目，显示原书的用字和拼音——目前的榕典这里显示的依然是推荐用字，得改掉——因为像戚林八音这样的来源，我们不可能把他的用字给改掉。

這個思路不錯。大標題我們可以用推薦用字，但是每個詞條的小標題可以用回原書用字。

> 为了方便用户查阅例句，我们将每个来源里面的用字默认显示为推荐用字

這處我覺得邏輯不太順。要不要**原書例句也一起默認按照原書用字來顯示**了，但是提供一個按鈕可以把詞條小標題和原書例句一鍵轉換爲榕典推薦用字？
榕典未來是有“例句”分頁的，原本打算在例句分頁顯示我們[cdo-corpus](https://github.com/MindongLab/cdo-corpus)的語料，以詞爲單位匹配。那這樣的話，我們也可以把馮/李書的例句扔進來一起參與檢索，並以「推薦用字」顯示？→ #977

**Reply by ZySieng** [reply]
我觉得不能这么做，第一是，把原书注释里面的字都改回去了，那就变成推荐用字一个词孤零零地悬在页面顶上，下面跟他一点关系都没有，等于推荐用字的意义被大大地弱化了。而且，这样一改，不只是这个条目的用字会变掉，注释里面的整个正字法都会变，到时候用户连在看的是哪个词都不知道——这么举例，我是个新手，在看“儥”的页面，除了榕典条目标题的这个“儥”，我全文看下来找不到一个“儥”字，冯书来源显示的是“勿会”，这“勿会”下面的例句里又有一个“无绽”，也是在搜索页面搜不出来的词，那我在这几分钟内到底看了什么？
第二是，这个{,}括号还有校对原书的作用，原书的错别字等一并是用这个格式改掉的，把他取消了，等于这些审校也都无效了。
把冯书李书更改为推荐用字后的例句扔进例句分页里，我觉得没什么意义。

**Reply by Guanchishan** [reply]
也行。
扔進例句欄有利於這種情況：
屬於甲詞條的某個例句包含了乙詞彙，則在乙的例句分頁同時顯示該例句。這樣有利於**提高例句的使用效率**

**Reply by ZySieng** [reply]
原来是这样子，那我也觉得放进例句有必要。
这个例句，到时候会按照义项来分类吗？还有就是，这个例句来源是什么样的，有没有收录其他文献的例句，如果是这样，保存数据时候是遵照原写法还是直接转写成了推荐用字？

**Reply by Guanchishan** [reply]
这个话题属于其他功能了，之后我会[另开一个issue](https://github.com/MindongLab/yngdieng/discussions/974)。

我也希望例句能够以义项为单位分类。这非常理想，但是这需要做到两点：
1. 需要把zingzeu_words里的福州话词汇进行析解义项的工作，以义项为单位建立新的ID；可以先以冯版词条为准绳，划割出义项来
1. 需要析出li、feng的例句，赋予ID
然后例句ID就可以和义项ID匹配了。可以鼓励用户来帮忙匹配。没匹配的仍然可以显示在例句分页，只要不要显示到某个具体义项下就好了。

目前除了冯李以外，只有音视频字幕、民间福州话文章和一些零散的句子，来源非常复杂，且第一手的很多。对于有刊载于书面文献的例句，我都对异体字模仿榕典做了花括号标记，以配合以后可能的显示原书设计。

我觉得目前例句分页可以一概显示推荐用字，有什么问题以后再说。

**Comment by ztl8702** [comment]
所以第一步先实现：
* 原书（冯版、李版）解释部分显示原书用字（含按钮）
* 原书注音按原书注音系统

？

**Reply by ZySieng** [reply]
嗯，再加上个每个来源的条目用字默认显示为原书内容。
与此同时，我们通过已经匹配的ID，人工校对“修改后的feng-zingzeuwords”的用字和拼音是否一致，
默认显示哪个音，要在zingzeuwords里面标记。←这个目前有没有特定的格式？V=之类的？

**Reply by ztl8702** [reply]
> 默认显示哪个音，要在zingzeuwords里面标记。←这个目前有没有特定的格式？V=之类的？

`YNGDIENG=true `

```
ABCD 某词
     a213 a242 V=市区 SC=连 YNGDIENG=true  #榕典优先显示
     ba53 a242 V=市区 SC=连
```

可先标注。暂未实现。（实现很容易）

**Comment by Guanchishan** [comment]
ztl said：榕典網頁版和微信小程序的用字顯示方式是不同的機制，爲解決本issue問題，要先解決前述兩套機制不統一的問題

**Reply by ztl8702** [reply]
This is already done.

**Comment by ztl8702** [comment]
一个问题：

按照目前的做法，戚林的“烟” huong1 是应该显示在 榕典 熏 条目 下 ，

但是 *同时* 是否显示在 “烟” 条目下？ （这个 huong1 算不算 戚林 给“烟”的训读音？）

**Reply by Guanchishan** [reply]
目前榕典似乎會把不同讀音視為不同詞彙。所以應該不用同時顯示在ieng條目下

**Reply by Guanchishan** [reply]
還是說你指的是搜索漢字得出的結果裡要不要包括houng55？

**Reply by ZySieng** [reply]
在熏条目下。语言是第一的，文字是派生的。一个条目下面显示的是共享同一个ID的来源，之所以是同一个ID，是因为他们的音义相同，跟文字没有关系。

**Reply by ztl8702** [reply]
OK

**Comment by Guanchishan** [comment]
按照前几天开会的结果，例句的～用字也要显示原版
还要显示原书拼音（显示原书拼音是否需要再开一个issue）
